util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,20 @@
+package com.twitter.util
+
+/** Provides a clean, lightweight interface for working with batched Futures */
+private[util] class Batcher[In, Out](
```

Comment:
this class can't be private if we're exposing it as a return type for Future.batched.  can you just make the constructor private instead?  That would look like:

```scala
class Batcher[In, Out] private[util](...
```